### ISSUE TITLE
RDP Wrapper and Windows system registry

### ISSUE THREAD

**Issue by SergeBuinenstern** [stascorp/rdpwrap#2489]
I am not a user or fan of RDP Wrapper. I installed it bundled with some VPN, I think Proton VPN. After reviewing Proton VPN, I deleted it. I also removed RDP Wrapper.
A few days later I discovered some unpleasant surprises:
1. Windows Defender now has undeletable exceptions.
2. The "Settings" applet in the "Credentials" section freezes for several minutes when moving between subsections.
After several consultations, I fixed the first problem using the Windows Registry. But it was not possible to cope with the second problem. If anyone can tell me which branches and registry parameters I need to look at, I will be very grateful.

**Comment by sashaqwert** [comment]
> 2\. The "Settings" applet in the "Credentials" section freezes for several minutes when moving between subsections.
>         After several consultations, I fixed the first problem using the Windows Registry. But it was not possible to cope with the second problem.

Try `sfc /scannow` (More in Microsoft website: https://support.microsoft.com/en-us/windows/using-system-file-checker-in-windows-365e0031-36b1-6031-f804-8fd86e0ef4ca )

> But it was not possible to cope with the second problem.

Reinstall Windows from original ISO...

**Comment by SergeBuinenstern** [comment]
Thanks. I have already visited the recommended site, used dism and sfc/scannow. This did not produce any results because no files were damaged, only some registry values were changed. Unfortunately, the uninstallation of the RDP-Wrapper was not organized correctly and did not provide for their return to their previous values. I didn’t think to save a restore point before my “experiments” or track the changes I made, I confess. When I realized that the problem was in the registry, I found the branch responsible for the Windows Defender policy and found exceptions there that I could not delete (they were restored after deletion). I fixed the registry and Windows Defender worked without any problems. I understand that the RDP-Wrapper has changed branches related to user policies (maybe to increase its rights or for some other purpose), but I do not have enough knowledge to determine which bushes would be worth comparing with similar ones in an initially correct machine . Since I have no chance of identifying this from open source code, I was hoping that people familiar with the RDP-Wrapper installation process could point me in the direction of my search. I was wrong, sorry.
I will not reinstall the system, because the inconvenience of freezing for 3-4 minutes when switching from the “Your data” option to any other is not too critical for me. Otherwise the system works flawlessly. Thank you!

**Comment by sashaqwert** [comment]
> only some registry values

RDP Wrapper only override the `termsrv.dll` library. Original (1.6.2) is not change other keys.

Defender exclusions path included to Auto update from Telegram (not included in original and mod).

**Comment by SergeBuinenstern** [comment]
Thank you. I realized that I was faced with a version of RDP-Wrapper modified for specific purposes, and it would not be possible to find the author for clarification. I devoted several hours to checking my operating system using Windows Defender, as well as several utilities, such as utilities from Kaspersky, DrWeb. The irony of fate: I was looking for a VPN just to bypass the blocks, because of which I had to contact malwarebytes.com through the Tor browser. In any case, no malicious, modified or suspicious files and processes were detected on my system. In the “Setting” applet, delays occur only in the “Accounts” item when accessing only two sub-items and amount to 3-4 minutes for the first access, and 0.5-2.0 minutes for all subsequent accesses. I haven't touched these settings in years, so I'll stop working on this problem after a few tests for additional vulnerabilities.
In this regard, I consider my question closed and apologize for the time I took up, and also thank Aleksandr Chivarzin for his comprehensive answer. Thank you!
